### PR TITLE
docs: backend tls policy

### LINKED ISSUE #2771: docs: Add docs for BackendTLSPolicy
*Description*:
>Describe the issue.

relates to https://github.com/envoyproxy/gateway/issues/2033

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

### PR DESCRIPTION
**Pull request by guydc**
**Which issue(s) this PR fixes**:
Fixes #2771

### CODE REVIEW

File: site/content/en/latest/user/security/backend-tls.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
+---
+title: "Mutual TLS: External Clients to the Gateway"
```
Comment (zhaohuabing) [inline site/content/en/latest/user/security/backend-tls.md:2]: Gateway to Backend Services ?

Comment (zhaohuabing) [reply site/content/en/latest/user/security/backend-tls.md:2]: And Mutal TLS or TLS?

Comment (guydc) [reply site/content/en/latest/user/security/backend-tls.md]: yea. my bad, it was a copy-paste mistake from the MTLS doc. fixed it now.

**Review by zhaohuabing — APPROVED** [review]
LGTM Thanks!

**Review by zirain — APPROVED** [review]

**Pull request merged**